### PR TITLE
Feat/component/checkbox

### PR DESCRIPTION
**Pull request by harrisazmi**
Component for checkbox

### CODE REVIEW

File: packages/react/src/components/checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        className,
+      )}
+      {...props}
+    >
+      {children}
+    </label>
+  );
+};
+
+CheckboxLabel.displayName = "CheckboxLabel";
```
Comment (irfancoder) [inline packages/react/src/components/checkbox.tsx:120]: I am looking to remove this. Including the label in `toggle.tsx`. Label will be a standalone component

Comment (harrisazmi) [reply packages/react/src/components/checkbox.tsx:120]: Roger boss. Noted.

File: packages/react/src/components/checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    size?: "small" | "medium" | "large";
+    disabled?: boolean;
+  }>
+> = ({ children, size = "small", disabled = false }) => {
+  return (
+    <CheckboxContext.Provider value={{ size, disabled }}>
+      {children}
+    </CheckboxContext.Provider>
+  );
+};
```
[the remark this answers is not in the document]
Comment (harrisazmi) [reply packages/react/src/components/checkbox.tsx:94]: Roger boss, Noted

File: packages/react/src/components/checkbox.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+import * as React from "react";
+import * as CheckboxPrimitive from "@radix-ui/react-checkbox";
+import { CheckIcon } from "../icons/check";
+import { MinusIcon } from "../icons/minus";
+import { clx } from "../utils";
+import { cva, type VariantProps } from "class-variance-authority";
+
+const checkboxVariants = cva(
+  [
+    "peer rounded-sm outline-none flex items-center justify-center",
```
[the remark this answers is not in the document]
Comment (harrisazmi) [reply packages/react/src/components/checkbox.tsx:11]: it was planned to be used for the label, but then i removed it. Now removing labels as needed.

File: packages/react/src/components/checkbox.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    "border data-[state!=checked]:border-otl-gray-200 text-txt-black-900 hover:border-gray-300 shadow-button",
```
[the remark this answers is not in the document]
Comment (harrisazmi) [reply packages/react/src/components/checkbox.tsx]: roger boss, updated

File: packages/react/src/components/checkbox.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    "data-[state=checked]:bg-primary-600 data-[state=checked]:border-none data-[state=checked]:hover:border-primary-700 data-[state=checked]:hover:bg-primary-700",
+    "data-[state=indeterminate]:bg-primary-600 data-[state=indeterminate]:border-none data-[state=indeterminate]:hover:border-primary-700 data-[state=indeterminate]:hover:bg-primary-700",
+    "focus-visible:ring-[3px] focus-visible:ring-fr-primary",
+    "focus:ring-[3px] focus:ring-fr-primary",
+    "disabled:opacity-50 disabled:cursor-not-allowed disabled:bg-bg-washed data-[state=checked]:disabled:bg-primary-600",
```
[the remark this answers is not in the document]
Comment (harrisazmi) [reply packages/react/src/components/checkbox.tsx]: yes boss, updated, following figma. i changed it initially cause its too dark, I cant see them if dark mode during checking and testing.

File: packages/react/src/components/checkbox.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
+"use client";
```
Comment (irfancoder) [inline packages/react/src/components/checkbox.tsx:1]: Remove this. This is Nextjs specific

File: packages/react/src/components/checkbox.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+const checkboxVariants = cva(
+  [
+    "rounded-sm outline-none flex items-center justify-center",
+    "border border-otl-gray-200 hover:border-otl-gray-300 shadow-button",
+    "data-[state=checked]:bg-primary-600 data-[state=checked]:border-none data-[state=checked]:hover:border-primary-700 data-[state=checked]:hover:bg-primary-700",
+    "data-[state=indeterminate]:bg-primary-600 data-[state=indeterminate]:border-none data-[state=indeterminate]:hover:border-primary-700 data-[state=indeterminate]:hover:bg-primary-700",
+    "focus-visible:ring-[3px] focus-visible:ring-fr-primary",
+    "focus:ring-[3px] focus:ring-fr-primary",
+    "disabled:opacity-30 disabled:cursor-not-allowed disabled:bg-bg-washed data-[state=checked]:disabled:bg-primary-600",
```
Comment (irfancoder) [inline packages/react/src/components/checkbox.tsx:17]: Missing `bg-bg-white`. Currently it is transparent

**Review by irfancoder — CHANGES_REQUESTED** [review]

File: packages/react/src/components/checkbox.tsx (hunk 6 above)
Comment (harrisazmi) [reply packages/react/src/components/checkbox.tsx:1]: roger boss

File: packages/react/src/components/checkbox.tsx (hunk 7 above)
Comment (harrisazmi) [reply packages/react/src/components/checkbox.tsx:17]: roger boss, done

**Review by irfancoder — APPROVED** [review]

**Pull request merged**